cpp/src/parquet/encoding.cc

```diff
@@ -2107,9 +2107,12 @@ class DictByteArrayDecoderImpl : public DictDecoderImpl<ByteArrayType>,
 
 template <typename DType>
 class DeltaBitPackEncoder : public EncoderImpl, virtual public TypedEncoder<DType> {
+  static_assert(std::is_same_v<int32_t, typename DType::c_type> ||
+                std::is_same_v<int64_t, typename DType::c_type>);
   // Maximum possible header size
   static constexpr uint32_t kMaxPageHeaderWriterSize = 32;
-  static constexpr uint32_t kValuesPerBlock = 128;
+  static constexpr uint32_t kValuesPerBlock =
+      std::is_same_v<int32_t, typename DType::c_type> ? 128 : 256;
   static constexpr uint32_t kMiniBlocksPerBlock = 4;
```

Comment:
Just to make sure I understand, is this where the 128 / 256 are coming from here?
```suggestion
  static constexpr uint32_t kMiniBlocksPerBlock = 4;
  static constexpr uint32_t kValuesPerBlock = kMiniBlocksPerBlock * sizeof(typename DType::c_type);
```

Reply:
Yes

Reply:
Although kValuesPerBlock is exactly equal to `kMiniBlocksPerBlock * sizeof(typename DType::c_type);`. This doesn't means they have some calculated relationship. So I hardcord it